vshard/replicaset.lua

```diff
@@ -170,7 +170,8 @@ end
 --
 local function replicaset_connect_to_replica(replicaset, replica)
     local conn = replica.conn
-    if not conn or conn.state == 'closed' then
+    -- gh-341: additionally check if connection's fiber was explicitly killed
+    if not conn or conn.state == 'closed' or util.is_conn_fiber_killed(conn) then
```

Comment:
Lets turn the check to be faster on the happy-path. I.e. if the connection is fine and dandy, then you shouldn't check an error message. It is done on each request, hence we want it to be as quick as possible in most cases:
```Lua
if conn and conn.state == 'active' then
    return conn
end
-- Do more rigorous checks
-- ...
```

Reply:
Done